sklearn/feature_selection/from_model.py

```diff
@@ -111,3 +135,98 @@ def transform(self, X, threshold=None):
             return X[:, mask]
         else:
             raise ValueError("Invalid threshold: all features are discarded.")
+
+
+class SelectFromModel(BaseEstimator, SelectorMixin):
+    """Meta-transformer for selecting features based on importance weights.
+
+    Parameters
+    ----------
+    estimator : object
+        The base estimator from which the transformer is built.
+        This can be both a fitted or a non-fitted estimator.
+        If it a fitted estimator, then ``transform`` can be called directly,
+        otherwise train the model using ``fit`` and then ``transform`` to do
+        feature selection.
+
+    threshold : string, float, optional
+        The threshold value to use for feature selection. Features whose
+        importance is greater or equal are kept while the others are
+        discarded. If "median" (resp. "mean"), then the ``threshold`` value is
+        the median (resp. the mean) of the feature importances. A scaling
+        factor (e.g., "1.25*mean") may also be used. If None and if
+        available, the object attribute ``threshold`` is used. Otherwise,
+        "mean" is used by default.
+
+    Attributes
+    ----------
+    `estimator_`: an estimator
+        The base estimator from which the transformer is built.
+        This is stored only when a non-fitted estimator is passed to the
+        ``SelectFromModel``.
+
+    `threshold_`: float
+        The threshold value used for feature selection.
+    """
+    def __init__(self, estimator, threshold=None):
+        self.estimator = estimator
+        self.threshold = threshold
+
+    def _get_support_mask(self):
+        # SelectFromModel can directly call on transform.
```

Comment:
Did we really decide this? I get why this might be helpful, but it breaks all sorts of conventions, e.g. if estimator is not fitted we should be raising `NotFittedError`. Note `CalibratedClassifierCV` supports pre-fitted estimators, but only with an explicit parameter `cv='prefit'`.

I would rather we have a standard way to freeze estimators for inclusion in various contexts. I have proposed such a helper class, but it and some code that supports it (#5080) doesn't seem to garner much interest.

In this case I think that this class is really not very interesting except for its use in a pipeline or similar. If one wants an ad-hoc mask over the `feature_importances_` attribute, it's not especially hard to get right.